localstack-core/localstack/services/lambda_/event_source_mapping/pollers/dynamodb_poller.py

```diff
@@ -84,14 +85,28 @@ def transform_into_events(self, records: list[dict], shard_id) -> list[dict]:
             dynamodb = record["dynamodb"]
 
             if creation_time := dynamodb.get("ApproximateCreationDateTime"):
-                # validated by TestDynamoDBEventSourceMapping.test_dynamodb_event_filter
+                # Float conversion validated by TestDynamoDBEventSourceMapping.test_dynamodb_event_filter
                 dynamodb["ApproximateCreationDateTime"] = float(creation_time.timestamp())
             event = {
                 # TODO: add this metadata after filtering (these are based on the original record!)
+                #  This requires some design adjustment because the eventId and eventName depend on the record.
                 "eventID": record["eventID"],
                 "eventName": record["eventName"],
                 # record content
                 "dynamodb": dynamodb,
             }
             events.append(event)
         return events
+
+    def failure_payload_details_field_name(self) -> str:
+        return "DDBStreamBatchInfo"
+
+    def get_approximate_arrival_time(self, record: dict) -> float:
+        # TODO: validate whether the default should be now
+        # Optional according to AWS docs:
+        # https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_StreamRecord.html
+        # TODO: parse float properly if present from ApproximateCreationDateTime -> now works, compare via debug!
+        return record["dynamodb"].get("todo", datetime.utcnow().timestamp())
```

Comment:
yes because using `ApproximateCreationDateTime` causes a parity issue and it's unclear whether that's the right field 🤔 .

Sidenote: ESM v1 attempts to get `ApproximateArrivalTimestamp` (which never exist for DynamoDB [GetRecords](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_GetRecords.html)) and therefore always defaults to `datetime.datetime.utcnow()`